waspc/src/Wasp/Generator/DockerGenerator.hs

```diff
@@ -46,7 +45,7 @@ genDockerfile spec = do
           object
             [ "usingPrisma" .= not (null $ AS.getDecls @AS.Entity.Entity spec),
               "dbSchemaFileFromServerDir" .= SP.fromRelFile dbSchemaFileFromServerDir,
-              "nodeMajorVersion" .= show (SV.major $ getLowestNodeVersionUserAllows spec),
+              "nodeVersion" .= show (getLowestNodeVersionUserAllows spec),
```

Comment:
Using a major can create problems for users. For example:
- `getLowestNodeVersionUserAllows spec` returns `18.6.0`, which means "You need at least Node 18.6.0 to run my app"
- We only specify the major version for the Docker image (in this case, `18`). The base image can therefore include any version that's backwards compatible with `18.0.0`
- Let's say that we end up building an image with Node `18.3.0`. If the user's code relies on changes introduced after `18.3.0` and before `18.6.0`, the build can fail or behave unexpectedly.

Therefore, it's always best to stick with the user's exact desired version.

The only question is - how can we be sure of this specific image's availability in DockerHub.

Reply:
I guess, users will report if something breaks and then we'll have to think about it? 🤷 

I have a feeling that the Node.js images are built automatically on release of new Node.js images: https://github.com/nodejs/docker-node/blob/main/.github/workflows/automatic-updates.yml

Reply:
Had the same feeling but was too lazy to verify :upside_down_face:

That's great news if true.

I linked this conversation from the issue.